internal/storage/postgres/tree/get_test.go

```diff
@@ -0,0 +1,534 @@
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		trees, err := r.GetTreesByIDs(ctx, []int32{1, 2})
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, trees)
+	})
+}
+func TestTreeRepository_GetCenterPoint(t *testing.T) {
+	t.Run("should successfully calculate and parse the center point", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		ids := []int32{1, 2, 3}
+		centroids, err := suite.Store.CalculateGroupedCentroids(context.Background(), ids)
+		if err != nil {
+			t.Fatal(err)
+		}
+		g, errGeos := geos.NewGeomFromWKT(centroids)
+		if errGeos != nil {
+			t.Fatal(errGeos)
+		}
+
+		// when
+		lat, long, errRepo := r.GetCenterPoint(context.Background(), ids)
+
+		// then
+		assert.NoError(t, errRepo)
+		assert.Equal(t, g.X(), lat, "Latitude does not match")
+		assert.Equal(t, g.Y(), long, "Longitude does not match")
+	})
+
+	t.Run("should return error if ids list is empty", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+
+		// when
+		lat, long, err := r.GetCenterPoint(context.Background(), []int32{})
+
+		// then
+		assert.Error(t, err)
+		assert.Equal(t, 0.0, lat)
+		assert.Equal(t, 0.0, long)
+	})
+}
+
+func TestTreeRepository_GetByTreeClusterID(t *testing.T) {
+	t.Run("should return trees successfully for a given tree cluster ID", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		treeClusterID := int32(1)
+		expectedTrees := testTrees[:2]
+
+		// when
+		trees, err := r.GetByTreeClusterID(context.Background(), treeClusterID)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, trees)
+		assert.NotEmpty(t, trees)
+		assert.Len(t, trees, len(expectedTrees))
+		for i, tree := range trees {
+			assertExpectedEqualToTree(t, expectedTrees[i], tree)
+		}
+	})
+
+	t.Run("should return empty list when no trees are linked with the given tree cluster ID", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+
+		// when
+		trees, err := r.GetByTreeClusterID(context.Background(), 99)
+
+		// then
+		assert.NoError(t, err)
+		assert.Empty(t, trees)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		trees, err := r.GetByTreeClusterID(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, trees)
+	})
+}
+
+func TestTreeRepository_GetByCoordinates(t *testing.T) {
```

Comment:
What about test case with negative latitude/longitude?

Reply:
I don't understand your question. What difference does it make if the latitude and longitude values are negative or positive? They are just numbers stored in the database.

Please have a look at the file  _**internal\entities\tree.go**_  to see the range validation of the latitude and longitude attributes. Checking these ranges is a part of the service layer.